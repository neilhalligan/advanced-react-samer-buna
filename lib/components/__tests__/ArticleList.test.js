import React from 'react';
import renderer from 'react-test-renderer';

import ArticleList from '../ArticleList';

const testProps = {
  articles: {
    a: { id: 'a' },
    b: { id: 'b' },
  },
  articleActions: {
    lookupAuthor: jest.fn(() => ({}))
  },
};

describe('ArticleList', () => {
  it('should render correctly', () => {
    const element = renderer.create(
      <ArticleList
        {...testProps}
      />
    ).toJSON();
    expect(element.children.length).toBe(2);
    expect(element).toMatchSnapshot();
  });
});
