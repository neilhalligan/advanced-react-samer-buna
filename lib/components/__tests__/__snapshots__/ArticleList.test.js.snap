// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList should render correctly 1`] = `
<div>
  <div>
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.8em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a>
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 10,
        }
      }
    />
  </div>
  <div>
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.8em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a>
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 10,
        }
      }
    />
  </div>
</div>
`;
